src/execution/execute.ts

```diff
@@ -922,6 +912,41 @@ function completeValue(
   );
 }
 
+async function completePromise(
```

Comment:
The name is a little bit too generic.
@yaacovCR How about `completePromisedValue`?

Reply:
Sounds great!

265e6873